### PR TITLE
[Impeller] one descriptor pool per frame.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Creating descriptor pools is expensive, no need to have more than one per frame.

The lifecycle of the descriptor pool is exactly the same as the cmd pool, which indicates that we can probably do some consolidation/refactoring in the future.

Fixes https://github.com/flutter/flutter/issues/157115

### CODE REVIEW

File: impeller/renderer/backend/vulkan/command_pool_vk.cc [hunk 1]
```diff
@@ -155,10 +156,6 @@ void CommandPoolVK::Destroy() {
   collected_buffers_.clear();
 }
 
-// Associates a resource with a thread and context.
-using CommandPoolMap =
```
Comment (jonahwilliams) [inline impeller/renderer/backend/vulkan/command_pool_vk.cc:159]: moved definition to header

File: impeller/renderer/backend/vulkan/command_pool_vk.cc [hunk 2]
```diff
@@ -169,17 +166,47 @@ using CommandPoolMap =
 //    available for reuse ("recycle").
 static thread_local std::unique_ptr<CommandPoolMap> tls_command_pool_map;
 
+struct WeakThreadLocalData {
```
Comment (jonahwilliams) [inline impeller/renderer/backend/vulkan/command_pool_vk.cc:169]: The thread local data now holds both the cmd pool and desc pool

File: impeller/renderer/backend/vulkan/command_pool_vk.cc [hunk 3]
```diff
@@ -293,14 +335,13 @@ void CommandPoolRecyclerVK::DestroyThreadLocalPools(const ContextVK* context) {
   Lock all_pools_lock(g_all_pools_map_mutex);
   auto found = g_all_pools_map.find(context);
   if (found != g_all_pools_map.end()) {
-    for (auto& weak_pool : found->second) {
-      auto pool = weak_pool.lock();
-      if (!pool) {
-        continue;
+    for (auto& [command_pool, desc_pool] : found->second) {
+      const auto& strong_pool = command_pool.lock();
```
Comment (jonahwilliams) [inline impeller/renderer/backend/vulkan/command_pool_vk.cc:339]: We don't actually need to do anything with the descriptor pool here, I just added it for symmetry....

**Review by matanlurey — APPROVED** [review]

**Comment by jonahwilliams** [comment]
reason for revert: failing on scenario test

**Pull request merged**